jina/executors/__init__.py

```diff
@@ -241,30 +240,52 @@ def config_abspath(self) -> str:
         """
         return self.get_file_from_workspace(f'{self.name}.yml')
 
+    @staticmethod
+    def get_shard_workspace(workspace_folder: str, workspace_name: str, pea_id: int) -> str:
+        # TODO (Joan, Florian). We would prefer not to keep `pea_id` condition, but afraid many tests rely on this
+        return os.path.join(workspace_folder, f'{workspace_name}-{pea_id}') if pea_id > 0 else os.path.join(workspace_folder)
```

Comment:
couldn't it be just `... else workspace_folder`?

Reply:
And furthermore, if the whole line would just be
```
if pea_id < 0:
    pea_id = 0
return os.path.join(workspace_folder, f'{workspace_name}-{pea_id}')
```
we might be able to heavily simplify the parsers, non? If we have shards, it makes total sense, that we have the `pea_id` in the workspace name. But if we don't have shards but only one `Pea`, it still makes total sense to add the `pea_id` to the workspace name (in my opinion).

Reply:
> couldn't it be just `... else workspace_folder`?

oh sure!!

Reply:
About the second point, I agree, but I was afraid that it would break current logic and a huge amount of tests, and thus we decided to keep it like this as in the `TODO` is expressd

Reply:
Ah I should read the comments as well... That's why I almost never write comments: I also never read them :D

Fine I'll approve